ray-operator/controllers/ray/rayservice_controller.go

```diff
@@ -191,16 +249,87 @@ func (r *RayServiceReconciler) updateState(ctx context.Context, rayServiceInstan
 	return err
 }
 
-func (r *RayServiceReconciler) getOrCreateRayClusterInstance(ctx context.Context, rayServiceInstance *rayv1alpha1.RayService) (*rayv1alpha1.RayCluster, error) {
-	// Update ray cluster
-	var rayClusterInstanceName string
+func (r *RayServiceReconciler) reconcileRayCluster(ctx context.Context, rayServiceInstance *rayv1alpha1.RayService) (*rayv1alpha1.RayCluster, *rayv1alpha1.RayCluster, error) {
+	rayClusterList := rayv1alpha1.RayClusterList{}
+	filterLabels := client.MatchingLabels{common.RayServiceLabelKey: rayServiceInstance.Name}
+	var err error
+	if err = r.List(ctx, &rayClusterList, client.InNamespace(rayServiceInstance.Namespace), filterLabels); err != nil {
+		return nil, nil, err
+	}
+
+	// Clean up RayCluster instances.
+	for _, rayClusterInstance := range rayClusterList.Items {
+		if rayClusterInstance.Name != rayServiceInstance.Status.RayClusterName && rayClusterInstance.Name != rayServiceInstance.Status.PreparingRayClusterName {
```

Comment:
I like the idea of keeping the current active RayCluster name in the status field.